ci-operator/step-registry/cucushift/upgrade/toversion/cucushift-upgrade-toversion-commands.sh

```diff
@@ -0,0 +1,492 @@
+            break
+        fi
+    done
+    if (( retry == 0 )); then
+        echo "Timeout to get recommended update!" 
+        return 1
+    fi
+}
+
+# judge if upgrade with --to or --to-latest cmd
+# set correct target_version for the upgrade
+function set_target_and_upgrade_cmd(){
+    mapfile -t available_versions < <(echo "$(oc get clusterversion version -o json|jq -r '.status.availableUpdates[]?.version')")
+    if [[ -z "${available_versions[*]}" ]]; then
+        echo "No recommended update available!"
+        exit 1
+    fi
+    # x.y.z: run upgrade with "oc adm upgrade --to x.y.z";
+    if [[ -n "${UPGRAGE_TO_VERSION}" ]] && [[ "${UPGRAGE_TO_VERSION}" != "latest" ]]; then
+        valid_target="false"
+        # check if x.y.z is in available_versions, if not, it's invalid target with --to.
+        for version in ${available_versions[*]}; do
+            if [[ "${UPGRAGE_TO_VERSION}" == "${version}" ]]; then
+                TARGET_VERSION="${UPGRAGE_TO_VERSION}"
+                valid_target="true"
+                upgrade_cmd="oc adm upgrade --to ${TARGET_VERSION}"
+                break
+            fi
+        done
+        if [[ "${valid_target}" == "false" ]]; then
+            echo "Specified target version ${UPGRAGE_TO_VERSION} is not in recommended update, upgrade will not continue!"
+            exit 1
+        fi
+    # get target_version from available updates
+    else
+        TARGET_VERSION=$(oc get clusterversion version -o json|jq -r '.status.availableUpdates[]?.version'|sort|tail -n1)
+        # there will be three types version in candidate channel:4.15.0-ec1,4.15.0-rc1,4.15.0
+        # when there are all above, sort will return 4.15.0-rc1 as latest(unexpected, should be 4.15.0),
+        # when there is both 4.15.0 and 4.15.0-rc1, sort will return 4.15.0-rc1 as latest(unexpected, should be 4.15.0),
+        # when there is both 4.15.0-rc1 and 4.15.0-ec1, sort will return 4.15.0-rc1 as latest(expected),
+        # there will not be 4.15.0 and 4.15.0-ec1 only, ignore
+        # so here we need correct unexpected latest with 4.15.0
+        if [[ "${TARGET_VERSION}" =~ "-" ]]; then
+            expected_target="${TARGET_VERSION%-*}"
+            for version in ${available_versions[*]}; do
+                if [[ "${version}" == "${expected_target}" ]]; then
+                    TARGET_VERSION="${expected_target}"
+                    break
+                fi
+            done
+        fi
+        # latest: run upgrade with "oc adm upgrade --to-latest";
+        if [[ "${UPGRAGE_TO_VERSION}" == "latest" ]]; then
+            upgrade_cmd="oc adm upgrade --to-latest"
+        # "": run upgrade with "oc adm upgrade --to $latest_version_in_available_update"
+        else
+            upgrade_cmd="oc adm upgrade --to ${TARGET_VERSION}"
+        fi
+    fi
+    TARGET_MINOR_VERSION="$(echo "${TARGET_VERSION}" | cut -f2 -d.)"
+    target_repo="${DUMMY_TARGET%:*}"
+    TARGET="${target_repo}:${TARGET_VERSION}"
```

Comment:
```suggestion
    TARGET=$(oc get clusterversion version -o json| jq -r --arg z "${TARGET_VERSION}" '.status.availableUpdates[]? | select(.version==$z) | .image')
```
Based on we already got `TARGET_VERSION` from CV's availableUpdates, I would suggest to get the target image from CV, too. Reduce interact with `DUMMY_TARGET`.